source/arm7/ieee_802_11/beacon.c

```diff
@@ -394,7 +394,7 @@ void Wifi_ProcessBeaconOrProbeResponse(Wifi_RxHeader *packetheader, int macbase)
                 {
                     // min rssi is 2, heh.
                     int tmp = packetheader->rssi_ & 255;
-                    for (int j = 0; j < 7; j++)
+                    for (int j = 0; j < 8; j++)
```

Comment:
Can we move `8` to a macro (`#define AP_RSSI_PAST_ENTRIES 8` or something) instead of it being a magic number?

In addition, there's code to "shift past measurements by one slot" below; I think it may be slightly more appropriate to implement the averaging as a ring buffer.

Reply:
A ring buffer would increase RAM usage, and the CPU cost of this shift is very small because it happens only once per beacon. I don't think that's necessary.

Reply:
That's fair.